actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ShadowUserMigrationScheduler.java

```diff
@@ -0,0 +1,312 @@
+     * @return shawdow user
+     */
+    private ShadowUser getUpdatedShadowUser( Map<String,Object>compositeKeysMap){
+        Response response=cassandraOperation.getRecordsByCompositeKey(JsonKey.SUNBIRD, JsonKey.SHADOW_USER,compositeKeysMap);
+        ProjectLogger.log("ShadowUserMigrationScheduler:getUpdatedShadowUser: record status in cassandra for getting the updated shawdow user object ".concat(response.getResult() + ""), LoggerEnum.INFO.name());
+        Map<String,Object>resultmap=((List<Map<String, Object>>)response.getResult().get(JsonKey.RESPONSE)).get(0);
+        ShadowUser shadowUser=mapper.convertValue(resultmap,ShadowUser.class);
+        return shadowUser;
+    }
+
+
+    private void updateStatusInUserBulkTable(String processId, int statusVal) {
+        try {
+            ProjectLogger.log("ShadowUserMigrationScheduler:updateStatusInUserBulkTable: got status to change in bulk upload table".concat(statusVal + ""), LoggerEnum.INFO.name());
+            Map<String, Object> propertiesMap = new HashMap<>();
+            propertiesMap.put(JsonKey.ID, processId);
+            propertiesMap.put(JsonKey.STATUS, statusVal);
+            updateBulkUserTable(propertiesMap);
+        } catch (Exception e) {
+            ProjectLogger.log("ShadowUserMigrationScheduler:updateStatusInUserBulkTable: status update failed".concat(e + ""), LoggerEnum.ERROR.name());
+        }
+    }
+
+    private void updateBulkUserTable(Map<String, Object> propertiesMap) {
+        Response response = cassandraOperation.updateRecord(bulkUploadDbInfo.getKeySpace(), bulkUploadDbInfo.getTableName(), propertiesMap);
+        ProjectLogger.log("ShadowUserMigrationScheduler:updateBulkUserTable: status update result".concat(response.getResult() + ""), LoggerEnum.INFO.name());
+    }
+
+    private void updateMessageInBulkUserTable(String processId, String key, String value) {
+        Map<String, Object> propertiesMap = new HashMap<>();
+        propertiesMap.put(JsonKey.ID, processId);
+        propertiesMap.put(key, value);
+        if (value.equalsIgnoreCase(JsonKey.SUCCESS)) {
+            propertiesMap.put(JsonKey.STATUS, ProjectUtil.BulkProcessStatus.COMPLETED.getValue());
+            propertiesMap.put(JsonKey.FAILURE_RESULT, "");
+        } else {
+            propertiesMap.put(JsonKey.SUCCESS_RESULT, "");
+            propertiesMap.put(JsonKey.STATUS, ProjectUtil.BulkProcessStatus.FAILED.getValue());
+        }
+        updateBulkUserTable(propertiesMap);
+    }
+
+
+    /**
+     * this method will take descision wheather to update the record or not.
+     * @param shadowUser
+     * @param migrationUser
+     * @return boolean
+     */
+    private boolean isSame(ShadowUser shadowUser,MigrationUser migrationUser){
+        if(!shadowUser.getName().equalsIgnoreCase(migrationUser.getName())){
+        return false;
+        }
+        if(StringUtils.isNotBlank(migrationUser.getEmail()) && !shadowUser.getEmail().equalsIgnoreCase(migrationUser.getEmail())){
```

Comment:
for shadow user email is optional , so what will happen if some user won't have  email. 
shadowUser.getEmail() : code will throw NPE

Reply:
@manzarul  we are taking them as a blank string it won't effect